lib/daemon_runner/session.rb

```diff
@@ -0,0 +1,124 @@
+require 'diplomat'
+
+module DaemonRunner
+  #
+  # Manage distributed locks with Consul
+  #
+  class Session
+    include Logger
+
+    class << self
+      attr_reader :session
+
+      def start(name, **options)
+        @session ||= Session.new(name, options).renew!
+      end
+
+
+      # Acquire a lock with the current session, or initialize a new session
+      #
+      # @param path [String]  A path in the Consul key-value space to lock
+      # @param lock_session [Session] The Session instance to lock the lock to
+      # @return [Boolean] `true` if the lock was acquired
+      #
+      def lock(path)
+        Diplomat::Lock.acquire(path, session.id)
```

Comment:
This is a really old comment that I wrote Thursday night but never finished. I want to preserve it for posterity in case you take a look at it:

It doesn't look like there's an exception handling in Diplomat if the path is wrong or session id is invalid.

You're wrapping the call to `Diplomat::Session.renew` in a `begin`/`rescue` block and that method looks like it works the same way as `Diplomat::Lock.acquire`.